### DISCUSSION TITLE
Is there a way to get the table name from schema?

### DISCUSSION THREAD

**Discussion by hasezoey** [diesel-rs/diesel#3700] [Q&A] [answered]
I know that there is `schema::table::columns::column::NAME` (trait [`Column::NAME`](https://docs.rs/diesel/latest/diesel/prelude/trait.Column.html#associatedconstant.NAME)) constant, but i couldnt find something for a given table, does something like this exist?

i would like to use this for something like errors without having to manually specify the table name anywhere or have to keep it in-sync with diesel

**Comment by weiznich** [comment, marked as answer]
We do not expose this information explicitly, but I think you can get that information via `diesel::debug_query(&your_table::table)`.

**Reply by hasezoey** [reply]
then as a feature-request, could it be added similar to what `Column::NAME` does?

**Reply by weiznich** [reply]
I don't think that's easily possible, as you need to somehow incorporate the schema name there as well. If you have a concrete proposal how to do that given the current API I might be open for it. Otherwise that's at least for me personally nothing that has priority, which means I likely won't spend time working (or arguing) about it.
